src/effects/effectsmanager.cpp

```diff
@@ -414,27 +420,30 @@ void EffectsManager::processEffectsResponses() {
                it.key() == response.request_id) {
             EffectsRequest* pRequest = it.value();
 
-            if (!response.success) {
-                qWarning() << debugString() << "WARNING: Failed EffectsRequest"
-                           << "type" << pRequest->type;
-            } else {
-                //qDebug() << debugString() << "EffectsRequest Success"
-                //           << "type" << pRequest->type;
-
-                if (pRequest->type == EffectsRequest::REMOVE_EFFECT_FROM_CHAIN) {
-                    //qDebug() << debugString() << "delete" << pRequest->RemoveEffectFromChain.pEffect;
-                    delete pRequest->RemoveEffectFromChain.pEffect;
-                } else if (pRequest->type == EffectsRequest::REMOVE_CHAIN_FROM_RACK) {
-                    //qDebug() << debugString() << "delete" << request->RemoveEffectFromChain.pEffect;
-                    delete pRequest->RemoveChainFromRack.pChain;
-                } else if (pRequest->type == EffectsRequest::REMOVE_EFFECT_RACK) {
-                    qDebug() << debugString() << "delete" << pRequest->RemoveEffectRack.pRack;
-                    delete pRequest->RemoveEffectRack.pRack;
-                }
-            }
+            // Don't check whether the response was successful here because
+            // specific errors should be caught with DEBUG_ASSERTs in
+            // EngineEffectsManager and functions it calls to handle requests.
+
+            collectGarbage(pRequest);
 
             delete pRequest;
             it = m_activeRequests.erase(it);
         }
     }
 }
+
+void EffectsManager::collectGarbage(const EffectsRequest* pRequest) {
+    if (pRequest->type == EffectsRequest::REMOVE_EFFECT_FROM_CHAIN) {
+        //qDebug() << debugString() << "delete" << pRequest->RemoveEffectFromChain.pEffect;
+        delete pRequest->RemoveEffectFromChain.pEffect;
+    } else if (pRequest->type == EffectsRequest::REMOVE_CHAIN_FROM_RACK) {
+        //qDebug() << debugString() << "delete" << request->RemoveEffectFromChain.pEffect;
+        delete pRequest->RemoveChainFromRack.pChain;
+    } else if (pRequest->type == EffectsRequest::REMOVE_EFFECT_RACK) {
+        //qDebug() << debugString() << "delete" << pRequest->RemoveEffectRack.pRack;
+        delete pRequest->RemoveEffectRack.pRack;
+    } else if (pRequest->type == EffectsRequest::DISABLE_EFFECT_CHAIN_FOR_INPUT_CHANNEL) {
+        //qDebug() << debugString() << "deleting states for input channel" << pRequest->channel << "for EngineEffectChain" << pRequest->pTargetChain;
+        pRequest->pTargetChain->deleteStatesForInputChannel(pRequest->channel);
```

Comment:
This needs to be changed to avoid race conditions.

Reply:
I suppose it is time to actually make use of the response side of the MessagePipe.

Reply:
Since the states are actually not stored in a Map, it probably works here. 
Please arrange a test, which disables a channel and enables it directly after. 
This can happen if a user is hammering in the enables switches, a controller mapping issue, or if no sound-card clock is available.

Reply:
How about this?

Reply:
I have manually tested it by just patching the code and it works. No unittests required.